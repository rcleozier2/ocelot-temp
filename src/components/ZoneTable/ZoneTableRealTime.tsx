import React from "react";
import { DataTable } from "primereact/datatable";
import { Column } from "primereact/column";
import { ColumnGroup } from "primereact/columngroup";
import { Row } from "primereact/row";
import { Assigned, Scheduled, Complete } from "../Icon/TableIcons";

interface Tasks {
  [key: string]: any;
}

interface Props {
  tasks: any;
  drivers: any;
}

const ZoneTableRealTime = (props: Props) => {
  const tableData: Array<object> = [];
  const tasks = props.tasks;

  const assigned = "Sche..";
  const failed = "Assig..";
  const justSend = "Done";

  const timeslots: Array<string> = [
    "slot4amTo9am",
    "slot9amTo12pm",
    "slot12pmTo3pm",
    "slot3pmTo6pm",
    "slot6pmTo9pm",
    "slot9pmTo4am"
  ];

  // Build Row Data
  tasks.data.forEach((task: any) => {
    let obj: Tasks = {
      name: task.name
    };

    timeslots.forEach(slot => {
      obj[slot + "_1"] = <Assigned data={task[slot]} />;
      obj[slot + "_2"] = <Scheduled data={task[slot]} />;
      obj[slot + "_3"] = <Complete data={task[slot]} />;
    });
    tableData.push(obj);
  });

  let headerGroup = (
    <ColumnGroup>
      <Row>
        <Column header="Zone" rowSpan={4} style={{ width: "120px" }} />
        <Column
          header="4AM - 9AM"
          colSpan={3}
          style={{ backgroundColor: "#F0E5D1" }}
        />
        <Column
          header="9AM-Noon"
          colSpan={3}
          style={{ backgroundColor: "#F4F4F4" }}
        />
        <Column
          header="Noon-3PM"
          colSpan={3}
          style={{ backgroundColor: "#F4F4F4" }}
        />
        <Column
          header="3PM-6PM"
          colSpan={3}
          style={{ backgroundColor: "#F4F4F4" }}
        />
        <Column
          header="6PM-9PM"
          colSpan={3}
          style={{ backgroundColor: "#F4F4F4" }}
        />
        <Column
          header="9PM-4AM"
          colSpan={3}
          style={{ backgroundColor: "#F0E5D1" }}
        />
      </Row>

      <Row>
        <Column header={assigned} />
        <Column header={failed} />
        <Column header={justSend} />

        <Column header={assigned} />
        <Column header={failed} />
        <Column header={justSend} />

        <Column header={assigned} />
        <Column header={failed} />
        <Column header={justSend} />

        <Column header={assigned} />
        <Column header={failed} />
        <Column header={justSend} />

        <Column header={assigned} />
        <Column header={failed} />
        <Column header={justSend} />

        <Column header={assigned} />
        <Column header={failed} />
        <Column header={justSend} />
      </Row>
    </ColumnGroup>
  );

  return (
    <>
    
      <DataTable value={tableData} headerColumnGroup={headerGroup}>
        <Column field="name" />

        <Column field="slot4amTo9am_1" className="scheduled" />
        <Column field="slot4amTo9am_2" className="assigned" />
        <Column field="slot4amTo9am_3" className="complete" />

        <Column field="slot9amTo12pm_1" className="scheduled" />
        <Column field="slot9amTo12pm_2" className="assigned" />
        <Column field="slot9amTo12pm_3" className="complete" />

        <Column field="slot12pmTo3pm_1" className="scheduled" />
        <Column field="slot12pmTo3pm_2" className="assigned" />
        <Column field="slot12pmTo3pm_3" className="complete" />

        <Column field="slot3pmTo6pm_1" className="scheduled" />
        <Column field="slot3pmTo6pm_2" className="assigned" />
        <Column field="slot3pmTo6pm_3" className="complete" />

        <Column field="slot6pmTo9pm_1" className="scheduled" />
        <Column field="slot6pmTo9pm_2" className="faassignediled" />
        <Column field="slot6pmTo9pm_3" className="complete" />

        <Column field="slot9pmTo4am_1" className="scheduled" />
        <Column field="slot9pmTo4am_2" className="assigned" />
        <Column field="slot9pmTo4am_3" className="complete" />
      </DataTable>
    </>
  );
};

export default ZoneTableRealTime;
