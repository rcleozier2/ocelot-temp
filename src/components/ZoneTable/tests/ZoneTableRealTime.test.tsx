import React from "react";
import ZoneTableRealTime from "../ZoneTableRealTime";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tasks = { data: [] };
  const drivers = { data: [] };

  const component = renderer
    .create(<ZoneTableRealTime tasks={tasks} drivers={drivers} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
