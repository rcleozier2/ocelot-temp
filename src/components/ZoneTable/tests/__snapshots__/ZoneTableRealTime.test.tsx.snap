// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="p-datatable p-component"
  id={null}
  style={null}
>
  <div
    className="p-datatable-wrapper"
  >
    <table
      className={null}
      style={null}
    >
      <thead
        className="p-datatable-thead"
      >
        <tr>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={4}
            style={
              Object {
                "width": "120px",
              }
            }
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Zone
            </span>
          </th>
          <th
            className=""
            colSpan={3}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={
              Object {
                "backgroundColor": "#F0E5D1",
              }
            }
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              4AM - 9AM
            </span>
          </th>
          <th
            className=""
            colSpan={3}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={
              Object {
                "backgroundColor": "#F4F4F4",
              }
            }
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              9AM-Noon
            </span>
          </th>
          <th
            className=""
            colSpan={3}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={
              Object {
                "backgroundColor": "#F4F4F4",
              }
            }
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Noon-3PM
            </span>
          </th>
          <th
            className=""
            colSpan={3}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={
              Object {
                "backgroundColor": "#F4F4F4",
              }
            }
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              3PM-6PM
            </span>
          </th>
          <th
            className=""
            colSpan={3}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={
              Object {
                "backgroundColor": "#F4F4F4",
              }
            }
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              6PM-9PM
            </span>
          </th>
          <th
            className=""
            colSpan={3}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={
              Object {
                "backgroundColor": "#F0E5D1",
              }
            }
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              9PM-4AM
            </span>
          </th>
        </tr>
        <tr>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Sche..
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Assig..
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Done
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Sche..
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Assig..
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Done
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Sche..
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Assig..
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Done
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Sche..
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Assig..
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Done
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Sche..
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Assig..
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Done
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Sche..
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Assig..
            </span>
          </th>
          <th
            className=""
            colSpan={null}
            onClick={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            rowSpan={null}
            style={null}
            tabIndex={null}
          >
            <span
              className="p-column-title"
            >
              Done
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        className="p-datatable-tbody"
      />
    </table>
  </div>
</div>
`;
